cubeb-backend/src/log.rs

```diff
@@ -23,8 +23,20 @@ macro_rules! cubeb_log_internal {
     };
     (__INTERNAL__ $msg: expr) => {
         if let Some(log_callback) = $crate::ffi::g_cubeb_log_callback {
-            let filename = std::path::Path::new(file!()).file_name().unwrap().to_str().unwrap();
-            let cstr = ::std::ffi::CString::new(format!("{}:{}: {}\n", filename, line!(), $msg)).unwrap();
+            use std::io::Write;
+
+            let mut buf = [0 as u8; 1024];
+            let filename = std::path::Path::new(file!())
+                .file_name()
+                .unwrap()
+                .to_str()
+                .unwrap();
+            // +2 for ':', +1 for ' ', and +1 for converting line value to number of digits
+            let len = filename.len() + ((line!() as f32).log10() as usize) + $msg.len() + 4;
+            assert!(len < buf.len(), "log is too long!");
```

Comment:
The assertion here is for an edge case. 

Without it, If the `len` is `1024`, `write!(...).unwrap()` can still work but `buf[len]` will hit an out-of-bounds error. To avoid that error, another way is to call `buf[cmp::min(len, buf.len()-1)] = 0` instead, which can silently trim the last character to a `nul` terminator in that case.

Reply:
Truncate the log message and make this a debug_assert!.  As it is, it seems likely we'll only trigger this "too late" once we're already trying to debug an issue in the field by enabling logging.